### PR TITLE
psr2 format and php7

### PR DESCRIPTION
**Pull request by totoleo**
* format by psr2;
* add function return type and fix some hints;

### CODE REVIEW

File: doc/Examples/01-passwords.php [hunk 1]
```diff
@@ -1,8 +1,8 @@
 <?php
-declare(strict_types=1);
-use \ParagonIE\Halite\HiddenString;
-use \ParagonIE\Halite\Password;
-use \ParagonIE\Halite\KeyFactory;
+declare(strict_types = 1);
```
Comment (sstok) [inline doc/Examples/01-passwords.php]: Actually the PSR "proposed" version doesn't use spaces 😄

File: doc/Examples/01-passwords.php [hunk 2]
```diff
@@ -1,8 +1,8 @@
 <?php
-declare(strict_types=1);
-use \ParagonIE\Halite\HiddenString;
```
Comment (sstok) [inline doc/Examples/01-passwords.php]: 👍

File: doc/Examples/01-passwords.php [hunk 3]
```diff
@@ -13,7 +13,7 @@
 }
 
 $password = new HiddenString('correct horse battery staple');
-$hash = Password::hash($password, $secretKey);
+$hash     = Password::hash($password, $secretKey);
```
Comment (sstok) [inline doc/Examples/01-passwords.php]: Please don't align these as they cause horrible diffs, phpdoc doesn't change that often and therefor is exceptional for this alignment.

File: src/Asymmetric/Crypto.php [hunk 4]
```diff
@@ -1,20 +1,12 @@
...
-    InvalidSignature
-};
 use ParagonIE\Halite\{
-    Util as CryptoUtil,
-    Halite,
-    HiddenString,
-    Key,
-    Symmetric\Crypto as SymmetricCrypto,
-    Symmetric\EncryptionKey
+    Halite, HiddenString, Key, Symmetric\Crypto as SymmetricCrypto, Symmetric\EncryptionKey, Util as CryptoUtil
```
Comment (sstok) [inline src/Asymmetric/Crypto.php]: 👎 less readable, secondly each use should be on it's own line.

**Comment by sstok** [comment]
👍 for stripping the white space.

File: doc/Examples/01-passwords.php (hunk 3 above)
Comment (totoleo) [inline doc/Examples/01-passwords.php]: @sstok  Yes, I know these aligns will cause many diffs, but I think I will help people read code easily.

File: src/Asymmetric/Crypto.php (hunk 4 above)
Comment (totoleo) [inline src/Asymmetric/Crypto.php]: @sstok  These imports usually will be eclipsed and that's why I make these lines join to one line.

Comment (totoleo) [inline src/Asymmetric/Crypto.php]: If you think it's not proper you can give me a format config, and I will import to my ide.

**Pull request closed without merging**